src/shogun/mathematics/graph/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+set(SHOGUN_GRAPH_HEADERS
+	Allocator.h
+	Graph.h
+	LinalgNodes.h
+	OperationImplementation.h
+	Shape.h
+	Tensor.h
+	Types.h
+	ops/Add.h
+	ops/Input.h
+	)
+
+set(SHOGUN_GRAPH_SOURCES
+	Graph.cpp
+	Tensor.cpp
+	ops/Add.cpp
+	ops/Input.cpp
+	)
+
+add_library(shogun-graph SHARED ${SHOGUN_GRAPH_HEADERS} ${SHOGUN_GRAPH_SOURCES} ${SHOGUN_CORE_HEADERS} ${CMAKE_BINARY_DIR}/src/shogun/lib/config.h)
+
+target_link_libraries(shogun-graph shogun-core)
```

Comment:
@vigsterkr as you can see I don't have great cmake skills.
Not sure if each folder/module should have it's own CMakeLists.txt?
Also I had to change the path of SHOGUN_CORE_HEADERS files to an absolute path

Reply:
yeah i was trying to do this myself with kernels, features but locally that was a failure, but imo that would the best that actually each subfolder has it's own cmakelists.txt